### PR TITLE
Added model factories to generate mocked data + 2 tests

### PR DESCRIPTION
**Pull request by ds161833**
now we can use neat factories to generate randomized data for testing

### CODE REVIEW

File: Client/src/Common/Helpers/DimensionHelpers.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { IDatasetModel } from "../../Models/Datasets/IDatasetModel"
+
+/**
```
Comment (ds161833) [inline Client/src/Common/Helpers/DimensionHelpers.tsx:3]: I extracted this from a component file

File: Client/src/Components/Graph/GraphControls/AxesControl.tsx [hunk 2]
```diff
@@ -64,6 +63,22 @@ export const getVariableDimension = (datasets: IDatasetModel[], variableName): n
   return Number(index)
 }
 
+const checkVariablesExist = (type: string, datasets: IDatasetModel[]): string[] => {
```
Comment (JeremiahTee) [inline Client/src/Components/Graph/GraphControls/AxesControl.tsx]: Nice

**Review by JeremiahTee — APPROVED** [review]
Nice changes, thanks for cleaning up the front-end unit tests and adding model factories
Build runs tests on both Client & Front-end & passes, amazing job

File: Client/src/Components/Search/SearchView.tsx [hunk 3]
```diff
@@ -21,7 +21,7 @@ export default function SearchView(props: IProps) {
   const [selection, setSelection] = useState<SelectionChangeParams>({ rowIds: [] })
 
   const handleSearchClick = async (query: ISearchDatasetsFormModel) => {
-    const datasets = await callGetDatasets(query)
+    const datasets = await callGetDatasets(query) || []
```
Comment (laithAwad) [inline Client/src/Components/Search/SearchView.tsx:24]: good catch, this would probably came as a bug later

**Review by laithAwad — APPROVED** [review]
Great work

**Pull request merged**